### PR TITLE
#638 - Move special VDOM behaviour into Incremental DOM overrides.

### PR DESCRIPTION
**Pull request by treshugart**
Implements #638. This also references https://github.com/google/incremental-dom/issues/281 as referenced in the issue for this. It would be better if we could have built-in support for "components" as described in the [babel-plugin-incremental-dom docs](https://github.com/jridgewell/babel-plugin-incremental-dom#components).

### CODE REVIEW

**Comment by lukebatchelor** [comment]
lgtm 👍

File: src/api/vdom.js [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
   skip,
   symbols,
-  text
-} = IncrementalDOM;
+  text,
+} from 'incremental-dom';
+import * as skateSymbols from './symbols';
+import { shadowDomV0, shadowDomV1 } from '../util/support';
 
+const fallbackToV0 = !shadowDomV1 && shadowDomV0;
```
Comment (joscha) [inline src/api/vdom.js]: should this be a global helper?

File: src/api/vdom.js [hunk 2]
```diff
@@ -43,6 +46,13 @@ attributes[symbols.default] = function (elem, name, value) {
     return name in elem ? applyProp(elem, name, value) : applyEvent(elem, name.substring(2), name, value);
   }
 
+  // Set the select attribute instead of name if it was a <slot> translated to 
+  // a <content> for v0.
+  if (name === 'name' && elem.tagName === 'CONTENT') {
+    name = 'select';
+    value = `[slot="${value}"]`;
```
Comment (joscha) [inline src/api/vdom.js]: do we need to encode value?

**Comment by joscha** [comment]
LGTM

File: src/api/vdom.js (hunk 2 above)
Comment (treshugart) [inline src/api/vdom.js]: I don't think so. They're allowed inside attribute values without escaping them.

File: src/api/vdom.js (hunk 1 above)
Comment (treshugart) [inline src/api/vdom.js]: I thought about that but decided against it since this is the only pace it's used. Happy to refactor later.

**Comment by joscha** [comment]
LGTM

**Pull request merged**